dipy/reconst/fwdti.py

```diff
@@ -0,0 +1,694 @@
+""" Classes and functions for fitting tensors without free water
+contamination """
+from __future__ import division, print_function, absolute_import
+
+import warnings
+
+import numpy as np
+
+import scipy.optimize as opt
+
+from .base import ReconstModel
+
+from dipy.reconst.dti import (TensorFit, design_matrix, _min_positive_signal,
+                              decompose_tensor, from_lower_triangular,
+                              lower_triangular, apparent_diffusion_coef)
+from dipy.reconst.dki import _positive_evals
+
+from dipy.core.sphere import Sphere
+from .vec_val_sum import vec_val_vect
+from dipy.core.ndindex import ndindex
+
+
+def fwdti_prediction(params, gtab, Diso=3.0e-3):
+    """ Signal prediction given the free water DTI model parameters.
+    Parameters
+    ----------
+    params : ndarray
+        Model parameters. The last dimension should have the 12 tensor
+        parameters (3 eigenvalues, followed by the 3 corresponding
+        eigenvectors) the volume fraction of the free water compartment, and
+        the non diffusion-weighted signal S0
+    gtab : a GradientTable class instance
```

Comment:
You should also check for at least 2 bvalues and just raise an error if the acquisition does not support FWE.